### ISSUE TITLE
Direction of the Arrows is Ambiguous, but Important

### ISSUE THREAD

**Issue by frohro** [bcpierce00/unison#506]
Hi,
I find myself always having to figure out which side my local computer is on for the synchronization.  I am memorizing that it is on the left, but this is not very clear in the GUI, and it is very important at times.  Is there a way you could make that less ambiguous?  Perhaps an icon or statement on the GUI that makes that perfectly clear?
Thanks for a useful tool, and thanks for thinking about this issue.
Rob

**Comment by tleedjarv** [comment]
Could you add a screenshot illustrating your issue? (Cut out any sensitive data.)
Are you using macOS native GUI or GTK+ GUI?

**Comment by frohro** [comment]
I am at school now, and I sync from home, so I can't provide you with a 
screenshot just now, but I'll do it this evening.  The issue happens 
when you are trying to decide whether left to right is from your local 
computer to the remote computer, or vice versa.  I am using the gnome 
gui, and there is no indication which is which.  I now know the local 
computer is on the left, and the remote is on the right, but it wasn't 
clear at first, and figuring that out wasn't easy because they were 
pretty similar.  The stakes are high too, because you could easily 
overwrite the wrong file. This happens when I forget to sync for a 
minute and save a file with a minor change, but at the other location I 
have done a lot of work I don't want to lose.

Thanks,

Rob

On 5/7/21 7:15 AM, Tõivo Leedjärv wrote:
> CAUTION: This email originated from outside the Walla Walla University 
> email system.
>
> Could you add a screenshot illustrating your issue? (Cut out any 
> sensitive data.)
> Are you using macOS native GUI or GTK+ GUI?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbcpierce00%2Funison%2Fissues%2F506%23issuecomment-834439962&data=04%7C01%7Crob.frohne%40wallawalla.edu%7C9010282cd9e94663bf4508d911629665%7Cd958f048e43142779c8debfb75e7aa64%7C0%7C0%7C637559937396092754%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=wSm0IVilvfUoVJ4YIvlwzBbXAFftkpNsWP7InVq070Y%3D&reserved=0>, 
> or unsubscribe 
> <https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAHS2BJQVGF5PQVSXYC4OETTMPYYRANCNFSM44JTSBVA&data=04%7C01%7Crob.frohne%40wallawalla.edu%7C9010282cd9e94663bf4508d911629665%7Cd958f048e43142779c8debfb75e7aa64%7C0%7C0%7C637559937396102750%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=10B0ef0opPrUU9Y2fDB3TB3tLXmfpZ0H24vOY47A6Pw%3D&reserved=0>.
>
-- 
Rob Frohne, Ph.D. P.E.
E. F. Cross School of Engineering
Walla Walla University
100 SW 4th Street
College Place, WA 99362
(509) 527-2075

**Comment by acolomb** [comment]
For the record, the order of the `root` options in the profile / on the command line determines which one is left / right in the gui.  AFAIK the first one ends up on the left side.  So you can actually control the order yourself to match your preference.

**Comment by adamryczkowski** [comment]
What would work, is the ability to label the roots in the config file. That improvement should be low-effort - just introduce an optional argument to the root, or a `rootlabel` command with a syntax similar to `rootalias`.

**Comment by fernan** [comment]
Hi, I know this has been tagged as 'wontfix' but I leave this here just in case it is considered in the future. 

Not only it is difficult to see or guess who is left or right (names are truncated and may be identical or very similar), at least in the Windows GTK2 GUI, the columns are not resizable, nor is there a tooltip nor any indication that pops up on placing the mouse over any of these columns. Furthermore, there is 

I am attaching a screenshot where my left (Dropbox) and my right (dropbox-may-2021) have been truncated in the GUI. In this particular example I can tell the difference (because of the upper vs lower 'D' in Dropbox), but you get the idea. 

![unison-gui-left-right](https://user-images.githubusercontent.com/175475/183466123-921e0bb8-bbe4-4b24-98f7-75faea7c5e25.png)
 
Just to reinstate, at least in the GUI: 
- names are truncated (can't tell easily which one is right or left)
- columns can't be resized (to show complete name of left or right targets)
- target names are not shown upon mouse over action (tooltips or similar)
- target names are also not shown in any of the dropdown menus

**Comment by gdt** [comment]
1) git master is now on gtk3.  We aren't going to address gtk2 anything.  I have no idea if this changes your results.

2) Feel free to figure out precisely how  it ought to be and implement it and submit a PR.  You have the code.  I don't mean to be nasty - but 1) this will get better if somebody makes it better and 2) you are welcome to change unison to make it do what you want and use it that way - the license says so and it's socially totally ok!

For discussion about the tradeoffs, I suggest unison-hackers@; the tracker  is intended for well-formed enhancement requests that are precise enough to implement.

**Comment by fernan** [comment]
Yes, sure, apologies for the noise then. Was just trying to help with what I can, but I get it, it is not enough, will try harder. Thanks for a great tool!

**Comment by gdt** [comment]
Your commenet was constructive - I am just trying to point out that things like this are only going to get fixed if someone who cares does the work, and that a change that works well for all is harder than one that works for you.  I sometimes carry changes like that privately and try to eventually make them general enough to send back.
